### PR TITLE
Account for v representing milliseconds introduced by RFC3339_EXTENDED

### PR DESCRIPTION
**Pull request by mariano**
This fixes PHP [bug 75577](https://bugs.php.net/bug.php?id=75577), with currently failing test submitted in [PHP PR 2495](https://github.com/php/php-src/pull/2945)

### CODE REVIEW

File: tests/files/bug75577.parseformat [hunk 1]
```diff
@@ -0,0 +1,4 @@
+TS: 0 | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.123
+TS: 0 | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.1234
+TS: 0 | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.12345
```
Comment (derickr) [inline tests/files/bug75577.parseformat]: Shouldn't anything with more than 3 characters fail with an error?

Comment (mariano) [reply tests/files/bug75577.parseformat]: I thought about that, but I did not see the `u` flag failing with more than 6 digits either. For example adding the following to `iso8601longtz.parseformat` succeeds:

```text
TS: 0 | -99999--99999--99999 01:00:03 0.123456 | H:i:s.u | 01:00:03.1234567 CET
```

Should I still work on ensuring `v` only takes up to 3 digits?

Comment (derickr) [reply tests/files/bug75577.parseformat]: Looks like it does actually fail if you run it manually: 

```
[GIT: pr/29]
derick@singlemalt:~/dev/timelib $ tests/tester-parse-string-by-format "Y-m-d H:i:s.v" "2010-03-06 16:07:25.12345"
E=1 TS: 0 | 2010-03-06 16:07:25 0.123000
Errors found while parsing 'Y-m-d H:i:s.v'
E Trailing data @ pos 23 (char=[4])
```

Just like with ``u``:

```
[GIT: pr/29]
derick@singlemalt:~/dev/timelib $ tests/tester-parse-string-by-format "Y-m-d H:i:s.u" "2010-03-06 16:07:25.1234567"
E=1 TS: 0 | 2010-03-06 16:07:25 0.123456
Errors found while parsing 'Y-m-d H:i:s.u'
E Trailing data @ pos 26 (char=[7])
```

And when running the tests with "make test-parse-format", it also shows this:

```
tests/files/bug75577.parseformat
OKAY | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.123
OKAY | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.1234 errors=1
OKAY | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.12345 errors=1
OKAY | 2010-03-06 16:07:25 0.123000 | Y-m-d H:i:s.v | 2010-03-06 16:07:25.123456 errors=1
```

So we're all good.

Can you please re-target this to the "bug fix" branch v2017 and squash your commits? Besides that, this looks good.

Comment (mariano) [reply tests/files/bug75577.parseformat]: @derickr done. Looks good?

**Comment by derickr** [comment]
Closing this, as it was already merged.

**Pull request closed without merging**